files/default/get_outbound_proxy.groovy

```diff
@@ -0,0 +1,47 @@
+import groovy.json.JsonOutput
+import org.sonatype.nexus.httpclient.config.ProxyServerConfiguration
+import org.sonatype.nexus.httpclient.config.UsernameAuthenticationConfiguration
+import org.sonatype.nexus.httpclient.config.NtlmAuthenticationConfiguration
+
+Map configMap(ProxyServerConfiguration psc) {
+    def config = [host: psc.getHost(), port: psc.getPort()]
+    def auth = psc.getAuthentication()
+    if (auth == null) {
+        return config
+    }
+
+    switch (auth.getType()) {
+        case UsernameAuthenticationConfiguration.TYPE:
+            def ua = (UsernameAuthenticationConfiguration)auth
+            config["auth"] = [username: ua.getUsername(), password: ua.getPassword()]
+            break
+        case NtlmAuthenticationConfiguration.TYPE:
+            def na = (NtlmAuthenticationConfiguration)auth
+            config["auth"] = [
+                    username: na.getUsername(),
+                    password: na.getPassword(),
+                    host: na.getHost(),
+                    domain: na.getHost(),
+            ]
+            break
+    }
+    return config
+}
+
+def existingConfig = [:]
+
+// From the UI you must first configure the HTTP proxy, then the HTTPS
```

Comment:
OPT: I would not return early if http is empty, I don't see the benefits

Reply:
Cleaner code to avoid extra indentation. if http is not set, nothing is set, that is a precondition.